timesketch/lib/stories/interface.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains the interface for a story exporter."""
+
+from __future__ import unicode_literals
+
+import json
+
+
+class StoryExporter(object):
+    """Interface for a story exporter."""
+
+    # String representing the output format of the story.
+    EXPORT_FORMAT = ''
+
+    def __init__(self):
+        """Initialize the exporter."""
+        self._data_lines = []
+        self._data_fetcher = None
+
+    @property
+    def data(self):
+        """Returns the data."""
+        return self.export_story()
+
+    def export_story(self):
+        """Export the story in the format."""
+        raise NotImplementedError
+
+    def from_string(self, json_string):
+        """Feed the exporter with a JSON string."""
+        blocks = json.loads(json_string)
+        if not isinstance(blocks, (list, tuple)):
+            return
+
+        for block in blocks:
+            self.from_dict(block)
+
+    def from_dict(self, block):
+        """Feed the exporter with a single block dict."""
+        component = block.get('componentName', 'N/A')
+        properties = block.get('componentProps', {})
+
+        if not self._data_fetcher:
+            return
+
+        if not component:
+            self._data_lines.append(block.get('content', ''))
+        elif component == 'TsViewEventList':
+            self._data_lines.append(
+                self._data_fetcher.get_view(properties.get('view')))
+        elif component == 'TsAggregationEventList':
+            self._data_lines.append(
+                self._data_fetcher.get_aggregation(
+                    properties.get('aggregation')))
+
+    def reset(self):
+        """Reset the story."""
```

Comment:
Be a little more descriptive here. What is this reset used for and when?

Reply:
better?